src/Rule/Components/VariableDefinition.php

```diff
@@ -4,18 +4,39 @@
 namespace PrinsFrank\Transliteration\Rule\Components;
 
 use PrinsFrank\Transliteration\Enum\Literal;
+use PrinsFrank\Transliteration\Exception\InvalidArgumentException;
+use PrinsFrank\Transliteration\FormalIdSyntax\Components\Character;
 use Stringable;
 
 final class VariableDefinition implements Stringable
 {
+    /**
+     * @throws InvalidArgumentException
+     *
+     * @phpstan-assert non-empty-string $name
+     * @phpstan-assert non-empty-string $value
+     */
     public function __construct(
         public readonly string $name,
         public readonly string $value,
     ) {
+        if ($value === '' || $name === '') {
+            throw new InvalidArgumentException('A variable cannot have an empty name or value');
+        }
+
+        if (Character::escape($name) !== $name) {
+            throw new InvalidArgumentException('A variable cannot contain any special characters');
+        }
     }
 
     public function __toString(): string
     {
-        return Literal::Dollar_Sign->value . $this->name . Literal::Space->value . Literal::Equals_Sign->value . Literal::Space->value . $this->value . Literal::Semicolon->value;
+        return Literal::Dollar_Sign->value
+               . $this->name
+               . Literal::Space->value
+               . Literal::Equals_Sign->value
+               . Literal::Space->value
+               . Character::escape($this->value)
+               . Literal::Semicolon->value;
```

Comment:
No art in code please.
```suggestion
            . $this->name
            . Literal::Space->value
            . Literal::Equals_Sign->value
            . Literal::Space->value
            . Character::escape($this->value)
            . Literal::Semicolon->value;
```

Reply:
Haha, this is somehow how my IDE thinks indenting should work. ;)